pkg/storegateway/gateway_blocks_http.go

```diff
@@ -156,9 +207,43 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 		ShowDeleted: showDeleted,
 		ShowSources: showSources,
 		ShowParents: showParents,
+		ActionType:  action,
 	}, blocksPageTemplate, req)
 }
 
+func (s *StoreGateway) performActionsOnBlocks(tenantID string, req *http.Request, action ActionType, blockUlids []string) error {
+	// When blockUlids is set, and dropdown action is "no-compact" or "delete-no-compact",
+	// we will perform the action on the selected blocks
+	if (action == ActionTypeNoCompact || action == ActionTypeDeleteNoCompact) && len(blockUlids) > 0 {
```

Comment:
we do the validation for the action and length of block IDs here, but also in `BlocksHandler`. Let's do it in one place only.

Reply:
note from @ying-jeanne the check for the empty string in BlocksHanlder and the check here are different - a non-empty string may still produce empty slice